blas/testblas/zaxpy.go

```diff
@@ -0,0 +1,119 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testblas
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+)
+
+type Zaxpyer interface {
+	Zaxpy(n int, alpha complex128, x []complex128, incX int, y []complex128, incY int)
+}
+
+func ZaxpyTest(t *testing.T, impl Zaxpyer) {
+	rnd := rand.New(rand.NewSource(1))
+
+	for tc, test := range []struct {
+		alpha complex128
+		x, y  []complex128
+
+		want    []complex128
+		wantRev []complex128
+	}{
+		{
+			alpha:   0,
+			x:       []complex128{2 + 3i, 3 + 4i, 4 + 5i},
+			y:       []complex128{20 + 30i, 30 + 40i, 40 + 50i},
+			want:    []complex128{20 + 30i, 30 + 40i, 40 + 50i},
+			wantRev: []complex128{20 + 30i, 30 + 40i, 40 + 50i},
+		},
+		{
+			alpha:   1,
+			x:       []complex128{2 + 3i, 3 + 4i, 4 + 5i},
+			y:       []complex128{20 + 30i, 30 + 40i, 40 + 50i},
+			want:    []complex128{22 + 33i, 33 + 44i, 44 + 55i},
+			wantRev: []complex128{24 + 35i, 33 + 44i, 42 + 53i},
+		},
+		{
+			alpha:   3 + 7i,
+			x:       []complex128{2 + 3i, 3 + 4i, 4 + 5i},
+			y:       []complex128{20 + 30i, 30 + 40i, 40 + 50i},
+			want:    []complex128{5 + 53i, 11 + 73i, 17 + 93i},
+			wantRev: []complex128{-3 + 73i, 11 + 73i, 25 + 73i},
+		},
+	} {
+		n := len(test.x)
+		for _, incX := range []int{-10, -1, 1, 10} {
+			for _, incY := range []int{-10, -1, 1, 10} {
+				aincX := incX
+				if aincX < 0 {
+					aincX = -aincX
+				}
+				aincY := incY
+				if aincY < 0 {
+					aincY = -aincY
+				}
+
+				x := make([]complex128, (n-1)*aincX+1)
+				for i := range x {
+					x[i] = complex(rnd.NormFloat64(), rnd.NormFloat64())
+				}
+				for i, v := range test.x {
+					x[i*aincX] = v
+				}
+				xCopy := make([]complex128, len(x))
+				copy(xCopy, x)
+
+				y := make([]complex128, (n-1)*aincY+1)
+				for i := range y {
+					y[i] = complex(rnd.NormFloat64(), rnd.NormFloat64())
+				}
+				for i, v := range test.y {
+					y[i*aincY] = v
+				}
+
+				want := make([]complex128, (n-1)*aincY+1)
+				copy(want, y)
+				if incX*incY > 0 {
+					for i, v := range test.want {
+						want[i*aincY] = v
+					}
+				} else {
+					for i, v := range test.wantRev {
+						want[i*aincY] = v
+					}
+				}
+
+				impl.Zaxpy(n, test.alpha, x, incX, y, incY)
+
+				prefix := fmt.Sprintf("Case %v (incX=%v,incY=%v):", tc, incX, incY)
+
+				equal := true
```

Comment:
We had discussed this in the [major breaking changes thread](https://groups.google.com/forum/#!topic/gonum-dev/4oLmSvtDuZY) but didn't come to an agreement on a single package (`Vector/<type>`?) or mirrored packages (`floats` and `complexes`).  If we're going to start building out the complex blas routines, we should get the vector routines organized, too.